diff --git a/include/boost/core/detail/sp_thread_sleep.hpp b/include/boost/core/detail/sp_thread_sleep.hpp
index c3e8927..e000f5d 100644
--- a/include/boost/core/detail/sp_thread_sleep.hpp
+++ b/include/boost/core/detail/sp_thread_sleep.hpp
@@ -56,7 +56,7 @@ using boost::core::detail::sp_thread_sleep;
 
 #include <time.h>
 
-#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__)
+#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__) && !defined(__OHOS__) && !defined(__OPENHARMONY__)
 # include <pthread.h>
 #endif
 
@@ -67,7 +67,7 @@ namespace core
 
 inline void sp_thread_sleep() BOOST_NOEXCEPT
 {
-#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__)
+#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__) && !defined(__OHOS__) && !defined(__OPENHARMONY__)
 
     int oldst;
     pthread_setcancelstate( PTHREAD_CANCEL_DISABLE, &oldst );
@@ -85,7 +85,7 @@ inline void sp_thread_sleep() BOOST_NOEXCEPT
 
     nanosleep( &rqtp, 0 );
 
-#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__)
+#if defined(BOOST_HAS_PTHREADS) && !defined(__ANDROID__) && !defined(__OHOS__) && !defined(__OPENHARMONY__)
 
     pthread_setcancelstate( oldst, &oldst );
 
